endpoints/apiconfig.go

```diff
@@ -777,7 +796,16 @@ func validateField(v reflect.Value, t reflect.StructField) error {
 
 	isZero := v.Interface() == reflect.Zero(v.Type()).Interface()
 	if isZero && tag.required {
-		return fmt.Errorf("missing field %v", t.Name)
+		return fmt.Errorf("missing field %v in %v", t.Name, parent.Name())
+	}
+
+	if v.Kind() == reflect.Struct {
+		if v.CanAddr() && hasExportedFields(v) {
```

Comment:
Why `CanAddr`? I feel like this should just call `validateRequest` all the time and filter out non validable fields during the iteration above.